system-parachains/encointer/src/treasuries_xcm_payout.rs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2023 Encointer Association
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+//! `PayOverXcm` struct for paying through XCM and getting the status back.
+
+use crate::xcm_config::KsmLocation;
+use alloc::vec;
+use core::marker::PhantomData;
+use frame_support::traits::{tokens::PaymentStatus, Get};
+use sp_runtime::traits::TryConvert;
+use xcm::{latest::Error, opaque::lts::Weight, prelude::*};
+use xcm_builder::LocatableAssetId;
+use xcm_executor::traits::{ConvertLocation, QueryHandler, QueryResponseStatus};
+
+pub use pallet_encointer_treasuries::Transfer;
+
+// This is the value that has been queried from the Asset Hub Kusama runtime.
+// There is an integration test in `integration-tests/emulated/tests/encointer/encointer-kusama/
+// That verifies that this fee is correct and will catch fee changes in Asset-Hub Kusama
+pub const REMOTE_XCM_TRANSFER_REMOTE_EXECUTION_FEE: u128 = 12749033321;
```

Comment:
Do you agree that this is a safe approach? We hardcode this value, and there is a test in the integration tests that would detect if the fee computation changes in the asset hub.